enterprise/cmd/frontend/internal/dotcom/productsubscription/tokens_db.go

```diff
@@ -0,0 +1,66 @@
+package productsubscription
+
+import (
+	"context"
+	"encoding/hex"
+	"strings"
+
+	"github.com/keegancsmith/sqlf"
+
+	"github.com/sourcegraph/sourcegraph/internal/database"
+	"github.com/sourcegraph/sourcegraph/internal/hashutil"
+	"github.com/sourcegraph/sourcegraph/lib/errors"
+)
+
+// defaultRawAccessToken is currently just a hash of the license key.
+func defaultRawAccessToken(licenseKey []byte) []byte {
+	return hashutil.ToSHA256Bytes(licenseKey)
+}
+
+type dbTokens struct {
+	db database.DB
+}
+
+// SetAccessTokenSHA256 activates the value as a valid token for the license.
+// The value should not contain any token prefixes.
+func (t dbTokens) SetAccessTokenSHA256(ctx context.Context, licenseID string, value []byte) error {
+	query := sqlf.Sprintf("UPDATE product_licenses SET access_token_sha256=%s WHERE id=%s",
+		hashutil.ToSHA256Bytes(value), licenseID)
+	res, err := t.db.ExecContext(ctx, query.Query(sqlf.PostgresBindVar), query.Args()...)
+	if err != nil {
+		return err
+	}
+	nrows, err := res.RowsAffected()
+	if err != nil {
+		return err
+	}
+	if nrows == 0 {
+		return errLicenseNotFound
+	}
+	return nil
```

Comment:
Can be simplified using `basestore.ScanFirstInt`:
```suggestion
	query := sqlf.Sprintf("UPDATE product_licenses SET access_token_sha256=%s WHERE id=%s RETURNING id",
		hashutil.ToSHA256Bytes(value), licenseID)
	_, ok, err := basestore.ScanFirstInt(t.db.Query(ctx, query))
	if err != nil {
		return err
	}
	if !ok {
		return errLicenseNotFound
	}
	return nil
```